src/Driver/GHCup.hs

```diff
@@ -133,3 +139,48 @@ initAppState = do
       let e' = pack $ prettyHFError e
       runLogger $ logError e'
       pure $ Left e'
+
+data GHCupResult
+  = GHCupSuccess (Maybe Text)
+  | GHCupFailed Text
+  deriving (Show)
+
+installGHC :: AppState -> GHCupDownloads -> Version -> IO GHCupResult
+installGHC st ds v =
+  ( flip runReaderT st
+      . runResourceT
+      . runE
+      $ installGHCBin v GHCupInternal False []
+  )
+    >>= \case
```

Comment:
Does the usage of `(&)` help with getting rid of `flip`?

Reply:
I’m not quite sure how that would work… possibly `(runReader & st) . …`, but I’m not sure that’s any more readable. Also, I consider `flip runReaderT` a fairly common idiom for running a monad transformer.